### ISSUE TITLE
5.2.6 breaks progress bar when unicode characters are in message

### ISSUE THREAD

**Issue by grasmash** [symfony/symfony#40633]
**Symfony version(s) affected**: 5.2.6

**Description**
5.2.6 included [these changes to strlenWithoutDecoration()](https://github.com/symfony/symfony/compare/v5.2.5...v5.2.6#diff-c088db8cd42c64b3dd749a6954f5382f124c9e164560f8a789031d36ab48c801R128) which break the progress bar when a unicode character is used as the progress bar character.
```
$progress->setProgressCharacter('💧');
```

**How to reproduce**
This test case reproduces the issue: https://github.com/acquia/cli/blob/master/tests/phpunit/src/Commands/Pull/PullDatabaseCommandTest.php#L294

The progress bar length is incorrectly calculated and results in an error:
```
str_repeat(): Second argument has to be greater than or equal to 0
/home/travis/build/acquia/cli/vendor/symfony/console/Helper/ProgressBar.php:517
/home/travis/build/acquia/cli/vendor/symfony/console/Helper/ProgressBar.php:576
/home/travis/build/acquia/cli/vendor/symfony/console/Helper/ProgressBar.php:589
/home/travis/build/acquia/cli/vendor/symfony/console/Helper/ProgressBar.php:414
/home/travis/build/acquia/cli/vendor/symfony/console/Helper/ProgressBar.php:323
/home/travis/build/acquia/cli/src/Output/Spinner/Spinner.php:151
/home/travis/build/acquia/cli/src/Output/Checklist.php:42
/home/travis/build/acquia/cli/tests/phpunit/src/Commands/ChecklistTest.php:28
```

**Possible Solution**

**Additional context**

**Comment by grasmash** [comment]
I can confirm that changing `$progress->setProgressCharacter('💧');` to `$progress->setProgressCharacter('^');` resolves the issue. The scope of the bug impact is known.

**Comment by grasmash** [comment]
Yeah so this change: 
https://github.com/symfony/symfony/compare/v5.2.5...v5.2.6#diff-c088db8cd42c64b3dd749a6954f5382f124c9e164560f8a789031d36ab48c801R128-R130
```
        if (preg_match('//u', $string)) {
            return (new UnicodeString($string))->width(true);
        }
```
Actually causes `💧` to be measured as 2 characters rather than 1, which creates the bug. I believe this is unrelated to https://github.com/symfony/symfony/pull/40460 and is related only to this change to strlenWithoutDecoration().

**Comment by chalasr** [comment]
Thank you for the report and the PR. Fixed in #40698
